### PR TITLE
Last controller tests

### PR DESCRIPTION
**Pull request by Eyedema**
Tested getAllCourses and addStudentToCourse, added the noMoreInteractions call on some tests and edited the getAllStudents test to better represent what should happen in the code.

### CODE REVIEW

**Comment by Eyedema** [comment]
All 3 code smells are necessary: # 1 and # 2 are for TODO implementations, # 3 is for a "return null" instead of an empty Collection (List) on getAllCourses. The method is mocked so it's not necessary to implement anything else at the time.

**Comment by Eyedema** [comment]
Successfully added mutation tests for the AgendaController class.

**Comment by Eyedema** [comment]
Successfully added Travis CI caching mechanism.

**Comment by Eyedema** [comment]
Created the skeleton for testing AgendaService, next is to setup TransactionCode probably as it's needed for the tests (functional interfaces etc...).

File: src/main/java/repository/CourseRepository.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package repository;
+
+public class CourseRepository {
```
Comment (palai103) [inline src/main/java/repository/CourseRepository.java:3]: This class should an interface. The real implementation will be provided by the Mongo class

Comment (palai103) [inline src/main/java/repository/CourseRepository.java:3]: This class should an interface. The real implementation will be provided by the Mongo class

File: src/main/java/repository/StudentRepository.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package repository;
+
+public class StudentRepository {
```
Comment (palai103) [inline src/main/java/repository/StudentRepository.java:3]: Same here. This should be an interface

**Pull request merged**